### PR TITLE
Key Iterator & 1.0 compatibility

### PR DESCRIPTION
**Pull request by madsciencetist**
Builds on top of #18 with minor additional 0.7 compatibility fixes

Adds an iterator type over LMDB keys and a `keys` method to return one. To iterate over `Int` keys and `Float64` values,
```
for key in keys(cur, Int)
    value = get(txn, dbi, key, Float64)
```

### CODE REVIEW

**Comment by wildart** [comment]
You should target 0.7/1.0 and drop compatibility with lower versions.

File: test/runtests.jl [hunk 1]
```diff
@@ -1,4 +1,6 @@
-using Base.Test
+using Compat
+using Compat.Test
+push!(LOAD_PATH, joinpath(dirname(@__FILE__), "../src"))
```
Comment (wildart) [inline test/runtests.jl]: No need to modify `LOAD_PATH`.

File: REQUIRE [hunk 2]
```diff
@@ -1,2 +1,3 @@
 julia 0.6
 BinDeps
+Compat
```
Comment (wildart) [inline REQUIRE]: Drop `Compat` & target 0.7+

File: src/LMDB.jl [hunk 3]
```diff
@@ -1,10 +1,16 @@
 __precompile__(true)
 module LMDB
 
-    isdefined(:Docile) && eval(:(@document))
+    using Compat
+
+    if VERSION < v"0.7"
```
Comment (wildart) [inline src/LMDB.jl]: Remove any code incompatible with 0.7 in all files.

**Review by wildart — COMMENTED** [review]
Drop support of older Julia versions

**Comment by madsciencetist** [comment]
Finally figured out Travis/Pkg3 enough to get tests to pass

**Comment by madsciencetist** [comment]
I set `version` to `0.1.0` in the `Project.toml` file with the expectation that you will tag the code as `v0.1.0`. Let me know if you'd prefer a different version number (I just need it to be greater than 0.0.4)

**Comment by madsciencetist** [comment]
Fixes #7

**Comment by madsciencetist** [comment]
Thanks!

**Pull request merged**